test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -0,0 +1,73 @@
+// Flags: --expose-internals
+import { mustCall } from '../common/index.mjs';
+import esmLoaderModule from 'internal/modules/esm/loader';
+import assert from 'assert';
+
+const { ESMLoader } = esmLoaderModule;
+
+{
```

Comment:
Why this big block?

Reply:
😅 https://github.com/nodejs/node/pull/41975#pullrequestreview-882269397

Reply:
Okay, but until you have the other cases it doesn’t make much sense to have just one block. I would also argue that perhaps other cases should be in new files, as this one’s already pretty large.

Reply:
I was thinking it would make sense to keep them in the same file. There are already a lot of files in this directory, many of which SEEM redundant (which makes it difficult to figure out what might or should be where). If the file gets renamed at the same time as more cases are added, git will likely break the history.

I could see removing the block for now and add it back when the other cases are added.

I would prefer to keep them in the same file though, unless they differ a lot or become unruly. Could I persuade you to wait and see?

Reply:
I’ve already approved, so you can do whatever. Maybe just add a comment at the top of the block saying this is here because you anticipate adding more blocks in the future, and what this block is meant to test.